src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter/PageSaveTempDataPropertyFilter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class PageSaveTempDataPropertyFilter : SaveTempDataPropertyFilterBase
+    {
+        public PageSaveTempDataPropertyFilter(ITempDataDictionaryFactory factory)
+            : base(factory)
+        {
+            OriginalValues = new Dictionary<PropertyInfo, object>();
```

Comment:
Yeah, right now `OriginalValues` is initialized sort of piecemeal across a couple differnt methods depending on which you run first, I'll try initializing it in the base constructor and see if anything breaks.